core/src/main/java/org/infinispan/interceptors/impl/AbstractIracLocalSiteInterceptor.java

```diff
@@ -192,52 +190,15 @@ protected void visitNonTxKey(InvocationContext ctx, Object key, WriteCommand com
    @SuppressWarnings("unused")
    private void handleNonTxDataWriteCommand(InvocationContext ctx, DataWriteCommand command, Object rv, Throwable t) {
       final Object key = command.getKey();
-      if (!command.shouldReplicate(ctx, true) || skipEntryCommit(ctx, command, key)) {
+      if (!command.isSuccessful() || skipEntryCommit(ctx, command, key)) {
```

Comment:
So looking at this change, doesn't the check for SKIP_XSITE_BACKUP flag in RemoveCommand#shouldReplicate handle this?

Reply:
Hrmm..  I guess not for the originating node. Does that mean the test sometimes removes a value from one site where it doesn't exist there but does on the other?

Reply:
The flag `SKIP_XSITE_BACKUP` is checked in `BaseBackupInterceptor` and the decision to send is in `BaseBackupInterceptor` logic. 

Whether the flag is set or not is not important here; we need to invoke `setMetadataToCacheEntry` to store the version for conflict resolution. `shouldReplicate()` was returning `false` and the version was not set. The conflict resolution did not work and the last update wins.

Reply:
Would rearranging the `setMetadataToCacheEntry` call to always be invoked when it is successful but says not to replicate? Or does it also have to be replicated?

Reply:
it does not depend if it is replicated or not.
The only condition is the entry will be committed locally:
* when the node is the primary owner
* when the context is remote and it is the backup owner.